components/src/dynamo/global_planner/tests/unit/test_scale_request_handler.py

```
@@ -338,3 +339,594 @@ async def test_handler_blocking_mode(mock_runtime):
+@pytest.mark.asyncio
+async def test_asymmetric_per_worker_gpu_pair_within_tolerance(mock_runtime):
+    """Prefill=1 GPU/worker, Decode=2 GPU/worker. Paired transfer lands at
+    total=max+tol and is accepted under tolerance."""
+    handler = ScaleRequestHandler(
+        runtime=mock_runtime,
+        managed_namespaces=["default-my-dgd"],
+        k8s_namespace="default",
+        min_total_gpus=10,
+        max_total_gpus=10,
+    )
+    # prefill: 4 replicas × 1 GPU = 4; decode: 3 × 2 = 6; total = 10
```

Ruff (0.15.11)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)